### DISCUSSION TITLE
Questions on FAQ.md

### DISCUSSION THREAD

**Discussion by mdanish-kh** [microsoft/winget-pkgs#67207] [Q&A] [answered]
I had some queries on FAQ.md which I hope to get clarified here (and possibly get added to FAQ.md as well).

> The best way to get the AppsAndFeaturesEntries, or the ARP Entries, for an installer is to run the installer inside of the [Windows Sandbox](https://docs.microsoft.com/windows/security/threat-protection/windows-sandbox/windows-sandbox-overview).

- What if someone does not own the Pro version of Windows? Is there any other reliable alternative to fetch the ARP entries? If there isn't I would just prefer if it was explicitly mentioned in FAQ.md. I guess the common entries i.e. DisplayName, Publisher can be seen from Control Panel, but more specifically I want to know how I can get the AppsAndFeaturesEntries ProductCode? Which I believe is different from ProductCode under Installers entry. (and practically where it should be used as I've read [here](https://github.com/microsoft/winget-pkgs/blob/master/doc/manifest/schema/1.1.0/installer.md) it's only for completeness and better matching. Is there more to it? because I'm seeing this entry used more often nowadays)

- What to do with a package that does not have Control Panel entries at all? I've seen this behavior with TorBrowser and also recently with a package request #66921 that these packages after getting installed don't show up on the Control Panel at all. How do we get ARP entries for such packages? 

- This may be an old question, but I haven't seen this documented anywhere. How do we verify a package installs correctly on all the architectures that its installer.yaml file provides? I guess the brute-force way is to have multiple VMs setup and verify the install procedure on all of them? Is there a better solution? (not related to Windows Sandbox if possible)

**Comment by vedantmgoyal9** [comment]
cc @Trenly

**Comment by Trenly** [comment, marked as answer]
1) I’m not sure; I know win10 home doesn’t support it, but MSFT docs aren’t exactly clear as to win11 home does or does not; Product code under the Installer or under AppsAndFeaturesEntries actually do refer to the same field. It is used only for completeness and matching as far as I know. I know this can be found using MSIViewer and Orca for MSI installers, or by using powershell to dig through the msi’s database, but for a exe it isn’t really possible to know what the product code is without installing it and looking at the registry entries, unless you have access to the bundling code and/or logs

2) AppsAndFeaturesEntries can’t be added for applications which do not create any ARP entries

3) Multiple VMs are how most people do it, but the Validation Pipeline also performs this validation - with the notable exception that it cannot yet validate arm64 installers do install correctly. Unless you have issues with a package repeatedly failing validation, it is rare that you would need to validate every environment manually

**Reply by mdanish-kh** [reply]
1. Oh, but the confusion came from this PR #66948 when you suggested these changes 
![image](https://user-images.githubusercontent.com/88161975/180510579-e8c3cf7a-4c48-4ea1-bf13-75022d6ac255.png)
I couldn't find the same product code through the usual powershell command `get-wmiobject Win32_Product | Sort-Object -Property Name |Format-Table IdentifyingNumber, Name, LocalPackage -AutoSize`

2. So in that case, whatever the Marketing version is should be used for PackageVersion field? 

3. Oh right, thank you.

**Reply by mdanish-kh** [reply]
Oh, I figured it out

> Win32_Product will only return applications installed via Windows Installer. There are [many](https://en.wikipedia.org/wiki/List_of_installation_software#Windows) products used to assemble installers that don’t build Windows Installer packages. Any applications that use these non-Windows Installer packages for deployment won’t be returned when Win32_Product is queried.

[Source](https://xkln.net/blog/please-stop-using-win32product-to-find-installed-software-alternatives-inside/). Other applications installed via `.exe` will not be visible by this command. For that either use the sandbox script or follow the steps provided in the source or use [this command](https://github.com/microsoft/winget-pkgs/discussions/67207#discussioncomment-3218852).

**Comment by mdanish-kh** [comment]
For 1. , there is an alternative ([thanks to this blog](https://xkln.net/blog/please-stop-using-win32product-to-find-installed-software-alternatives-inside/)), one can use this command to fetch the common ARP entries of installed packages.

```bash
$32BitPath = "SOFTWARE\Wow6432Node\Microsoft\Windows\CurrentVersion\Uninstall\*";$64BitPath = "SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\*"; Get-ItemProperty "HKLM:\$32BitPath", "HKLM:\$64BitPath", "Registry::\HKEY_CURRENT_USER\$32BitPath", "Registry::\HKEY_CURRENT_USER\$64BitPath","Registry::\HKEY_USERS\$($_.SID)\$32BitPath", "Registry::\HKEY_USERS\$($_.SID)\$64BitPath", "Registry::\HKEY_USERS\temp\$32BitPath", "Registry::\HKEY_USERS\temp\$64BitPath"  | Select DisplayName, PSChildName, DisplayVersion, Publisher
```
